src/gen_rules.ml

```diff
@@ -154,7 +154,9 @@ module Gen(P : Install_rules.Params) = struct
         ~compile_info ~dir_kind =
     let obj_dir = Utils.library_object_directory ~dir lib.name in
     let requires = Lib.Compile.requires compile_info in
-    let dep_kind = if lib.optional then Build.Optional else Required in
```

Comment:
Maybe the optional field should just be `Lib_deps_info.Kind.t`?

Reply:
That would work, but the field shouldn't be called `optional` then